drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -778,6 +951,31 @@ local function occupancy_attr_handler(driver, device, ib, response)
   device:emit_event(ib.data.value == 0x01 and capabilities.motionSensor.motion.active() or capabilities.motionSensor.motion.inactive())
 end
 
+local function cumul_energy_exported_handler(driver, device, ib, response)
+  device:set_field(TOTAL_EXPORTED_ENERGY, ib.data.elements.energy.value)
+  device:emit_event(capabilities.energyMeter.energy({ value = ib.data.elements.energy.value, unit = "Wh" }))
+end
+
+local function per_energy_exported_handler(driver, device, ib, response)
+  local latest_energy_report = device:get_field(TOTAL_EXPORTED_ENERGY) or 0
+  local summed_energy_report = latest_energy_report + ib.data.elements.energy.value
+  device:set_field(TOTAL_EXPORTED_ENERGY, summed_energy_report)
+  device:emit_event(capabilities.energyMeter.energy({ value = summed_energy_report, unit = "Wh" }))
+end
+
+local function energy_report_handler_factory(is_cumulative_report)
+  return function(driver, device, ib, response)
+    if not device:get_field(EXPORT_POLL_TIMER_IS_SET) then
+      set_poll_report_timer_and_schedule(device, is_cumulative_report)
+    end
+    if is_cumulative_report then
+      cumul_energy_exported_handler(driver, device, ib, response)
+    elseif device:get_field(CUMULATIVE_REPORTS_NOT_SUPPORTED) then
```

Comment:
Just so I understand this, if a situation where a device supports cumulative reports but the driver receives a periodic report from the device, we would ignore it? Aka if a device supports cumulative reports then those are the only ones that we care about, right?

Reply:
Yes that is correct.

Reply:
Since both of these would be mapping to the same value, we do not want the possibility that two different systems are at at play, possibly against one another.

Reply:
Ok, makes sense!